### DISCUSSION TITLE
Transfering ownership of nixGL to nix-community

### DISCUSSION THREAD

**Discussion by guibou** [nix-community/infra#882] [Q&A] [answered]
Hello.

I'm https://github.com/guibou/nixGL author.

nixGL is used in multiple context where you want to use OpenGl/Vulkan with application built with nix on a system where the installed OpenGL driver is built with a different nixpkgs than the one used by your application (e.g. all non nixos system AND all nixos system when the installed package is from another version of nixpkgs).

I wrote nixGL a few years ago when I was working in the computer graphics industry and wanted to deploy OpenGL tools on multiples machines. At that time I had reasons to use nixGL everyday, and I also had access to Nvidia / Intel / AMD (and hybrid) hardware. So life as a nixGL maintainer was not too painful (e.g. nixGL is difficult to test because it depends on the hardware and installed software of the system on which it is run, and that's difficult / impossible to "emulate", so actually, testing nixGL needs to test on multiple machines with different hardware).

However, I left the computer graphics industry and I'm no longer using OpenGL applications. Or the one I use, I install them with nixos / home-manager and it works out of the box because I'm not in the context were nixGL is required. I have no access anymore to OpenGL specific hardware (I only own an integrated intel GPU now).

So maintaining nixGL is difficult for me, I have no motivation, no incentive, I cannot test anything (I was used to approve MR without testing / discussion) and I cannot help in the design / evolution of nixGL (for example, there are multiples MR waiting for review related to an integration with home-manager, or `wrapPkg`, ...). Sadly, nixGL is also the only opensource project I made which is kinda "successful" so I cannot just let it die and ignore the MR / issue which are stacking, it seems that some people relies on it and that it is an important brick in order to get nix adoption in a context were OpenGL/vulkan is used.

Could nixGL ownership be transferred to nix-community?

**Comment by zimbatm** [comment]
Hey @guibou :)

For sure, I invited you to the github org and then you can transfer the repo. Let us know when it's done so we can fix the permissions.

To find a new maintainer, I suggest asking the people who are sending the PRs, or just giving them access. It's actually surprising how well it works when we trust people to do good things.

**Reply by guibou** [reply, marked as answer]
Thank you @zimbatm (hi ;)

https://github.com/nix-community/nixGL looks good.

**Reply by aware70** [reply]
Hey @zimbatm @guibou  - would it be possible to address nix-community/nixGL#143 in light of this change?

@guibou still shows as the active maintainer, but without a license it seems this software is in a state of limbo.

**Reply by guibou** [reply]
@aware70 I'm really sorry by how things are (not) evolving with nixGL.

There is a (stale) MR here: https://github.com/nix-community/nixGL/pull/174 for licence.

I'm wondering what to do next in order to give more rights to user.

**Reply by zimbatm** [reply]
Usually, the procedure is to:
1. Pick a license that you like.
2. Create a list of all the significant authors.
3. Ask them if they will release their changes under that copyright (including yourself).
4. Add the license to the project.
